src/visualization/annotation/annotation_base.cpp

```diff
@@ -73,18 +94,19 @@ AnnotationBase::returnAnnotations(bool drop_null) {
   if (!drop_null) {
     annotation_global->annotation_sframe = copy_data;
     return copy_data;
-  }
+    }
 
-  std::vector<std::string> annotation_column_name = {m_annotation_column};
-  std::list<std::shared_ptr<unity_sframe_base>> dropped_missing =
-      copy_data->drop_missing_values(annotation_column_name, false, false, false);
+    std::vector<std::string> annotation_column_name = {m_annotation_column};
+    std::list<std::shared_ptr<unity_sframe_base>> dropped_missing =
```

Comment:
I'm not sure I understand the overall approach here. Why do we drop missing values at all, only to add them back in upon return? Wouldn't it be simpler to send all rows (including those with missing data) and receive all rows? We don't necessarily need to display those rows on the client, in order to send and receive them.

If this approach was considered and rejected, please add a comment here as to why; if not, I think we should consider it.

Reply:
Sure. It's explained just now in previous threads.

In short, we can do everything inline but it's tricky and requires non-trivial work. Maybe @abhishekpratapa can explain it more. So we designed this workaround with the consideration that annotation is done by humans and each batch job won't be that large (less than 1000 pics per batch) for them. So split the data set won't be the bottle neck.